bukkit/src/main/java/com/github/games647/fastlogin/bukkit/listener/protocollib/VerifyResponseTask.java

```diff
@@ -175,13 +176,31 @@ private boolean enableEncryption(SecretKey loginKey) throws IllegalArgumentExcep
             //get the NMS connection handle of this player
             Object networkManager = getNetworkManager();
 
-            //try to detect the method by parameters
-            Method encryptMethod = FuzzyReflection
-                    .fromObject(networkManager).getMethodByParameters("a", SecretKey.class);
-
-            //encrypt/decrypt following packets
-            //the client expects this behaviour
-            encryptMethod.invoke(networkManager, loginKey);
+            try {
+                //try to detect the method by parameters
+                Method encryptMethod = FuzzyReflection
+                        .fromObject(networkManager).getMethodByParameters("a", SecretKey.class);
+
+                //encrypt/decrypt following packets
+                //the client expects this behaviour
+                encryptMethod.invoke(networkManager, loginKey);
+            } catch (Exception ex) {
```

Comment:
Falling back on exceptions implies a very bad performance in my experience. The login process could be a very performance critical part, so it's the best to check the method availability once and then save the result. The best timing would be the initialization of the ProtocolLib module in my opinion.

Reply:
I've changed it so that the method reflections are static, so they will be initialized just once and reused